sql/tablespace.sql

```diff
@@ -0,0 +1,105 @@
+-- select_tablespace() is used to assign a tablespace to a chunk.  A
+-- tablespace is selected from a set of tablespaces associated with
+-- the chunk's hypertable, if any.
+CREATE OR REPLACE FUNCTION _timescaledb_internal.select_tablespace(
+    hypertable_id INTEGER,
+    chunk_id      INTEGER
+)
+    RETURNS NAME LANGUAGE PLPGSQL VOLATILE AS
+$BODY$
+<<main_block>>
```

Comment:
yes, it provides namespacing so that one can reference a declared variable `foo` with `main_block.foo`. This avoids conflicts with other `foo`s, for instance columns with that name.

https://www.postgresql.org/docs/9.6/static/plpgsql-structure.html